testapp/src/main/java/com/rakuten/tech/mobile/testapp/ui/display/preload/PreloadMiniAppWindow.kt

```diff
@@ -70,13 +86,48 @@ class PreloadMiniAppWindow(private val context: Context, private val preloadMini
             nameView.text = "No info found for this miniapp!"
         }
 
+        // set manifest/metadata to UI: permissions
+        val permissionAdapter = PreloadMiniAppPermissionAdapter()
+        preloadMiniAppLayout.findViewById<RecyclerView>(R.id.listPreloadPermission).layoutManager =
+            LinearLayoutManager(context)
+        preloadMiniAppLayout.findViewById<RecyclerView>(R.id.listPreloadPermission).adapter =
+            permissionAdapter
+        preloadMiniAppLayout.findViewById<RecyclerView>(R.id.listPreloadPermission)
+            .addItemDecoration(
+                DividerItemDecoration(context, DividerItemDecoration.VERTICAL)
+            )
+
+        // TODO: inflate UI from MiniApp.getRequiredCustomPermissions
+        // TODO: inflate UI from MiniApp.getOptionalCustomPermissions
+        val namesForAdapter: ArrayList<MiniAppCustomPermissionType> = arrayListOf()
+        val resultsForAdapter: ArrayList<MiniAppCustomPermissionResult> = arrayListOf()
+
+        GlobalScope.launch(Dispatchers.IO) {
```

Comment:
Can we do `getMiniAppManifest` in `ViewModel` where the container call, i.e `MiniAppListViewModel` and then pass the required parameters to `PreloadMiniAppWindow`?

Reply:
Yeah, I have updated with `ViewModel`.